registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/validator/WorkflowActionRequestValidator.java

```diff
@@ -0,0 +1,147 @@
+package io.mosip.registration.processor.reprocessor.validator;
+
+import java.util.Objects;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.util.DateUtils;
+import io.mosip.registration.processor.core.exception.WorkflowActionRequestValidationException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.workflow.dto.WorkflowActionDTO;
+
+
+/**
+ * The Class WorkflowActionRequestValidator.
+ */
+@Component
+public class WorkflowActionRequestValidator {
+
+	/** The Constant VER. */
+	private static final String VER = "version";
+	
+	/** The Constant TIMESTAMP. */
+	private static final String TIMESTAMP = "requesttime";
+
+	/** The Constant ID_FIELD. */
+	private static final String ID_FIELD = "id";
+
+	/** The Constant WORKFLOW_ACTION_ID. */
+	private static final String WORKFLOW_ACTION_ID = "mosip.registration.processor.workflow.action.id";
+
+	/** The Constant WORKFLOW_ACTION_VERSION. */
+	private static final String WORKFLOW_ACTION_VERSION = "mosip.registration.processor.workflow.action.version";
+
+	Logger regProcLogger = RegProcessorLogger.getLogger(WorkflowActionRequestValidator.class);
+
+	/** The env. */
+	@Autowired
+	private Environment env;
+
+	/**
+	 * Validate.
+	 *
+	 * @param workflowActionDTO the workflow action DTO
+	 * @param errors            the errors
+	 * @return true, if successful
+	 * @throws WorkflowActionRequestValidationException
+	 */
+	public boolean validate(WorkflowActionDTO workflowActionDTO)

```

Comment:
If validate throws an exception on validation errors, then lets not return boolean also, since it is redudant.